flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java

```diff
@@ -448,20 +457,24 @@ public void testTriggerSavepoint() throws Exception {
 			try {
 				JobID id = new JobID();
 				{
-					CompletableFuture<String> savepointPathFuture = restClusterClient.triggerSavepoint(id, null);
-					String savepointPath = savepointPathFuture.get();
+					String savepointPath = cancelJob
```

Comment:
If we only assert on the job cancel flag being set to `true`, a single test would be enough imo:

```
       @Test
	public void testCancelJobWithSavepointShouldTriggerSavepointWithCancelJobFlag() throws Exception {
		final boolean cancelJob = true;
		final String expectedSavepointLocation = "/savepoints/savepoint-0d2fb9-8d5e0106041a";

		final TestSavepointHandlers testSavepointHandlers = new TestSavepointHandlers();
		final TestSavepointHandlers.TestSavepointTriggerHandler triggerHandler =
			testSavepointHandlers.new TestSavepointTriggerHandler(
				cancelJob, "/savepoints");
		final TestSavepointHandlers.TestSavepointHandler savepointHandler =
			testSavepointHandlers.new TestSavepointHandler(new SavepointInfo(
				expectedSavepointLocation,
				null));

		try (TestRestServerEndpoint restServerEndpoint = createRestServerEndpoint(
			triggerHandler,
			savepointHandler)) {
			
			final RestClusterClient<?> restClusterClient = createRestClusterClient(restServerEndpoint.getServerAddress().getPort());
			final String savepointLocation = restClusterClient.cancelWithSavepoint(new JobID(), "/savepoints");
			assertEquals(expectedSavepointLocation, savepointLocation);
		}
	}
```

It was a mistake to have that many scenarios + asserts in `testTriggerSavepoint`. The test was hard to read to begin with and this change makes it even harder. 

If you think the current state is readable enough, maybe consider eliminating the code duplication:
```
	private static String triggerSavepoint(final boolean cancelJob, final String targetSavepointDirectory, final RestClusterClient<?> restClusterClient) throws Exception {
		return cancelJob
			? restClusterClient.cancelWithSavepoint(new JobID(), targetSavepointDirectory)
			: restClusterClient.triggerSavepoint(new JobID(), targetSavepointDirectory).get();
	}
```

Reply:
We have to test both scenarious (cancel being true and false), as otherwise we wouldn't detect bugs where the flag is _always_ set or parsed to true of false.

Reply:
I agree that the test is overloaded, but don't think this flag makes this significantly worse. I would be up for refactoring the test into 4-5 separate ones in a follow-up.

Reply:
I meant that a single test for testing that the flag can be set to `true` would be enough. For the `false` case all scenarios in `testTriggerSavepoint` should be kept.

Reply:
that's fair.